internal/resources/metal/helper/helper.go

```diff
@@ -0,0 +1,32 @@
+package helper
+
+import (
+	"fmt"
+	"path"
+
+	"github.com/equinix/equinix-sdk-go/services/metalv1"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+)
+
+func GetMetalSSHKeyOwnerId(key *metalv1.SSHKey) basetypes.StringValue {
+	owner, exists := key.AdditionalProperties["owner"]
+	if !exists {
+		fmt.Errorf("owner not found")
```

Comment:
We're not returning any errors from this function, so as currently written this helper will end up introducing the problem @displague is referring to rather than fixing it: rather than failing when the SDK changes, we will quietly change the organization ID from a UUID to an empty string.

We _could_ improve the helper by returning these errors and then checking for them in the caller, but IMO that's overkill and we should just read the owner value from AdditionalProperties and cast it to `map[string]interface{}` without checking for nil.  As mentioned in the other thread, when an attribute disappears from AdditionalProperties, blindly casting it will cause a panic.  That will be caught in the acceptance tests without the need for a helper, and the ugly code will motivate us to fix the issue rather than maintaining a helper after it's no longer necessary.